x-pack/test/alerting_api_integration/observability/threshold_rule/avg_pct_no_data.ts

```diff
@@ -0,0 +1,179 @@
+    const THRESHOLD_RULE_ALERT_INDEX = '.alerts-observability.threshold.alerts-default';
+    const ALERT_ACTION_INDEX = 'alert-action-threshold';
+    const DATA_VIEW_ID = 'data-view-id-no-data';
+    let actionId: string;
+    let ruleId: string;
+
+    before(async () => {
+      await createDataView({
+        supertest,
+        name: 'no-data-pattern',
+        id: DATA_VIEW_ID,
+        title: 'no-data-pattern',
+      });
+    });
+
+    after(async () => {
+      await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
+      await supertest.delete(`/api/actions/connector/${actionId}`).set('kbn-xsrf', 'foo');
+      await esClient.deleteByQuery({
+        index: THRESHOLD_RULE_ALERT_INDEX,
+        query: { term: { 'kibana.alert.rule.uuid': ruleId } },
+      });
+      await esClient.deleteByQuery({
+        index: '.kibana-event-log-*',
+        query: { term: { 'kibana.alert.rule.consumer': 'alerts' } },
+      });
+      await deleteDataView({
+        supertest,
+        id: DATA_VIEW_ID,
+      });
+    });
+
+    describe('Rule creation', () => {
+      it('creates rule successfully', async () => {
+        actionId = await createIndexConnector({
+          supertest,
+          name: 'Index Connector: Threshold API test',
+          indexName: ALERT_ACTION_INDEX,
+        });
+
+        const createdRule = await createRule({
+          supertest,
+          tags: ['observability'],
+          consumer: 'alerts',
+          name: 'Threshold rule',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          params: {
+            criteria: [
+              {
+                aggType: Aggregators.CUSTOM,
+                comparator: Comparator.GT,
+                threshold: [0.5],
+                timeSize: 5,
+                timeUnit: 'm',
+                customMetrics: [
+                  { name: 'A', field: 'system.cpu.user.pct', aggType: Aggregators.AVERAGE },
+                ],
+              },
+            ],
+            alertOnNoData: true,
+            alertOnGroupDisappear: true,
+            searchConfiguration: {
+              query: {
+                query: '',
+                language: 'kuery',
+              },
+              index: DATA_VIEW_ID,
+            },
+          },
+          actions: [
+            {
+              group: FIRED_ACTIONS_ID,
+              id: actionId,
+              params: {
+                documents: [
+                  {
+                    ruleType: '{{rule.type}}',
+                  },
+                ],
+              },
+              frequency: {
+                notify_when: 'onActionGroupChange',
+                throttle: null,
+                summary: false,
+              },
+            },
+          ],
+        });
+        ruleId = createdRule.id;
+        expect(ruleId).not.to.be(undefined);
+      });
+
+      it('should be active', async () => {
+        const executionStatus = await waitForRuleStatus({
```

Comment:
Kinda up to you, since you own these tests, but we generally treat each test as stand-alone, so that you could change `it(...` to `it.only(...` when diagnosing tests.  I think we'd more generally write a function here to create the rule / connector, and call it from each test, and then delete it (somehow - ObjectRemover or other) in an `afterEach()`.

Reply:
@pmuellr, thank you for the feedback! 
I wrote these tests as a file-per-scenario as we are sharing the same tests in serverless and stateful env to maintain them easily (copy/paste 😄) with minimum dependencies until we can share one test suite across ENVs (agnostic tests)

Reply:
Ya, will be interesting to see how serverless / stateful test sharing will go!

> maintain them easily (copy/paste 😄)

Probably more of a risk of these being problematic, in that each test depends on the previous to work.  Small number of tests here though, so hard to imagine it going wrong in this file, now.  Something to think about if these get more complex.